### ISSUE TITLE
[Bug] iOS - CollectionView EmptyView has incorrect bounds, height, width

### ISSUE THREAD

**Issue by MaxHampton** [xamarin/Xamarin.Forms#7251]
### Description

I have a PopupPage (Rg.Plugins.Popup) that is acting as a multiple selection dialog, and am using a CollectionView to display the items for multiple selection. One particular use case in the app I'm working on requires selection from about 300 items. On Android, these items render instantly, however on iOS (with the update to 4.2 to get rid of the empty item on bounce) it is taking many seconds to render. I've added an empty view to display a loading indicator while waiting for those items to render, however, the empty view isn't respecting the fact that it's in a dialog with margins, and instead behaves as though it thinks the CollectionView is taking up the entire screen. The activity indicator is set up with Horizontal and VerticalOptions set to Center. It is not centered in the CollectionView, though. Instead it is offset on the X coordinate by approximately the Dialog's margin, and on the Y coordinate by the Dialog's margin plus the height of the header row. When the items are loaded, the CollectionView looks fine, only the EmptyView is having trouble here.

### Expected Behavior

CollectionView's EmptyView should be constrained to the bounds of the CollectionView.

### Actual Behavior

CollectionView's EmptyView behaves as though it is taking up the full screen regardless of the CollectionView's dimensions.

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Screenshots

![IMG_0007 1](https://user-images.githubusercontent.com/4153653/63555364-7629e700-c4f5-11e9-88a4-03433d2f72d9.PNG)

**Comment by samhouts** [comment]
@MaxHampton  Can you please attach a small project that demonstrates this issue? Thanks!

**Comment by MaxHampton** [comment]
@samhouts Here's a small demo illustrating the issue. I built in a delay to make sure the EmptyView is displayed and discovered on Android that the Horizontal and Vertical options may not be behaving correctly.

https://github.com/MaxHampton/CollectionViewIssues

**Comment by luccasclezar** [comment]
A workaround for now is to name the layout of the CollectionView's EmptyView, like this:

```XAML
<CollectionView.EmptyView>
    <ContentView x:Name="EmptyViewLayout">
        <ActivityIndicator HeightRequest="56" WidthRequest="56" VerticalOptions="Center" HorizontalOptions="Center" IsRunning="True" IsVisible="True" />
    </ContentView>
</CollectionView.EmptyView>
```

<br>

Then on the page's constructor (change the `EmptyViewLayout` property to the name you set):

```C#
// Only needed because the Parent property is set after the constructor, so wait until it's set.
EmptyViewLayout.PropertyChanged += (sender, e) =>
{
    string propertyName = e.PropertyName;

    if (propertyName == nameof(Parent))
    {
        // The EmptyViewLayout.Parent is the CollectionView
        if (EmptyViewLayout.Parent is VisualElement parent)
        {
            // When the CollectionView's size changes, update the Bounds of the EmptyViewLayout.
            parent.SizeChanged += (parentSender, e2) =>
            {
                // If the Parent's size is set, run Layout() with the Parent's Width and Height.
                if ((parentSender as VisualElement).Height > 0)
                    EmptyViewLayout.Layout(new Rectangle(0, 0, (parentSender as VisualElement).Width, (parentSender as VisualElement).Height));
            };
        }
    }
};
```

<br>

That's it. Until the issue is fixed this workaround will do :).

The repro project with the workaround:

[CollectionViewIssues.zip](https://github.com/xamarin/Xamarin.Forms/files/3664657/CollectionViewIssues.zip)

**Comment by samhouts** [comment]
closed by #7811
